pallets/identity/src/auth.rs

```diff
@@ -157,6 +158,10 @@ impl<T: Config> Module<T> {
         auth_id: &u64,
     ) -> Option<Authorization<T::AccountId, T::Moment>> {
         Self::authorizations(target, *auth_id).filter(|auth| {
+            // check if count is zero
+            if auth.count == 0u32 {
+                return true;
+            }
```

Comment:
Seems not needed since auth.count=0 authorisations are automatically removed anyway,.